framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1042,7 +1042,19 @@ protected void processConditionalPostconditions(
     @Override
     public TransferResult<V, S> visitCase(CaseNode n, TransferInput<V, S> in) {
         S store = in.getRegularStore();
-        return new RegularTransferResult<>(finishValue(null, store), store);
+        TransferResult<V, S> result =
+                new ConditionalTransferResult<>(
+                        finishValue(null, store), in.getThenStore(), in.getElseStore(), false);
+
+        V caseValue = null;
+        if (n.getCaseOperand()
+                != null) { // When encountering a default node, getCaseOperand returns null.
```

Comment:
Ah, I can remove that. I experimented with actually adding a case node for the default case in `SwitchBuilder`, and needed to add that null check there. But since that didn't result in `default` blocks working correctly, I should've removed the null check.